File: scripts/python/test_browser_automation.py
import asyncio
from browser.browser_provider import default_browser_provider
from browser.url_checker import UrlChecker
from browser.screenshot import capture_screenshot

async def main():
    print("Testing Python Browser Automation...")
    checker = UrlChecker(default_browser_provider)
    urls = [
        "https://scholar.google.com",
        "https://www.example.com"
    ]

    for url in urls:
        print(f"\nChecking: {url}")
        result = await checker.check_url(url, timeout_s=15)
        print(f"  Live: {result.is_live}")
        print(f"  Status: {result.status_code}")
        print(f"  Resolved: {result.resolved_url}")
        print(f"  Paywall: {result.is_paywall}")
        print(f"  Title: {result.page_title}")

    print("\nCapturing screenshot of example.com...")
    await capture_screenshot(default_browser_provider, "https://www.example.com", "test-py-screenshot.png", timeout_s=15)

    await default_browser_provider.close()
    print("Done.")

if __name__ == "__main__":
    asyncio.run(main())

File: scripts/test-browser-automation.ts
import { defaultBrowserProvider } from '../lib/browser/BrowserProvider';
import { UrlChecker } from '../lib/browser/UrlChecker';
import { captureScreenshot } from '../lib/browser/Screenshot';

async function main() {
  console.log('Testing TypeScript Browser Automation...');
  
  const checker = new UrlChecker(defaultBrowserProvider);
  const urls = [
    'https://scholar.google.com',
    'https://www.example.com'
  ];

  for (const url of urls) {
    console.log(`\nChecking: ${url}`);
    const result = await checker.checkAsync(url, 15);
    console.log(`  Live: ${result.isLive}`);
    console.log(`  Status: ${result.statusCode}`);
    console.log(`  Resolved: ${result.resolvedUrl}`);
    console.log(`  Paywall: ${result.isPaywall}`);
    console.log(`  Title: ${result.pageTitle}`);
  }

  console.log('\nCapturing screenshot of example.com...');
  await captureScreenshot(defaultBrowserProvider, 'https://www.example.com', 'test-ts-screenshot.png', 15);

  await defaultBrowserProvider.close();
  console.log('Done.');
}

main().catch(console.error);
